hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/autoassign/AutoAssignChecker.java

```diff
@@ -119,12 +94,14 @@ public void check() {
      */
     private void checkByTargetFilterQueryAndAssignDS(final TargetFilterQuery targetFilterQuery) {
         try {
-            final DistributionSet distributionSet = targetFilterQuery.getAutoAssignDistributionSet();
-
             int count;
             do {
 
-                count = runTransactionalAssignment(targetFilterQuery, distributionSet.getId());
+                final List<String> controllerIds = targetManagement
+                        .findByTargetFilterQueryAndNonDS(PageRequest.of(0, PAGE_SIZE),
```

Comment:
I guess we can use Constants.MAX_ENTRIES_IN_STATEMENT as the page size as it is used later during partitioning in assignment